std/traits.d

```diff
@@ -5423,6 +5425,58 @@ unittest
 }
 
 /**
+ * Detects whether type $(D T) is a multi dimensional array.
+ * Params:
+ *      T = type to be tested
+ *
+ * Returns:
+ *      true if T is a multi dimensional array
+ */
+template isMultiDimensionalArray(T)
+{
+    static if (!isArray!T)
+        enum isMultiDimensionalArray = false;
+    else
+    {
+        alias DT = typeof(T.init[0]);
+        enum isMultiDimensionalArray = hasLength!DT || isNarrowString!DT;
+    }
+}
+///
+unittest
+{
+    assert(isMultiDimensionalArray!(string[]) );
+    assert(!isMultiDimensionalArray!(int[]) );
+    assert(isMultiDimensionalArray!(int[][]) );
```

Comment:
Nitpick: for doc readability, could you move the `asserts` that pass next to each other and the same for the ones that don't. It just makes it easier to scan docs.